builtin/providers/aws/structure.go

```diff
@@ -1276,3 +1277,139 @@ func flattenApiGatewayThrottleSettings(settings *apigateway.ThrottleSettings) []
 
 	return result
 }
+
+// TODO: refactor some of these helper functions and types in the terraform/helper packages
```

Comment:
yes, I just wanted to hide them here for now until I get a feel for how these work with other providers. These might help with the AWS API, but other guards may be needed with different API semantics, or some functionality might be better off provided by the existing schema types.